### PR TITLE
Updated parseInput validation to take both min and max into account

### LINKED ISSUE #4100: Textbox Opacity field allows invalid values
## Bug Description

While the opacity field allows only values between 0 and 100, the opacity field for the textbox (background) has no such validation.

## Expected Behaviour

Both field should sanitize input the same way

## Steps to Reproduce

1. Insert text element
1. Choose "highlight" fill mode
1. Change colors and opacity for both text and text background

## Screenshots

Screenshot 2020-08-19 at 17 20 44

## Additional Context

- Plugin Version: current `main`
- Operating System: macOS
- Browser: Chrome 84

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by mwritter**
## Context

Ensure that all `opacityInput`s are within the given range [0, 100] 

## Summary

Fixes an oversight in `input/validation.js`.  Takes into account the combination of the `min` and `max` options now.

## Relevant Technical Choices

`input/validation.js` was overwriting the return value incorrectly if the `min` and `max` options were both set instead of chaining the logic.

## User-facing changes

Before/After:
https://user-images.githubusercontent.com/14057928/165819621-25f1e468-3fbf-46ae-8aa9-1500450280f5.mp4

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Go to the Story Editor
2. Add components that have an opacity field (Text, Shape, Sticker, Image, etc).
3. Update the opacity field with any numbers outside the range of [0, 100] and press Enter or click outside the field

On Staging:
- Notice the opacity field allows for negative numbers but doesn't allow for numbers greater that 100

On Branch:
- Notice the opacity field doesn't allow for negative numbers, it will set the field to 0 if a negative number is entered and still doesn't allow for numbers greater than 100

### Does this PR have a security-related impact?
No

### Does this PR change what data or activity we track or use?
No

### Does this PR have a legal-related impact?
No

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #4100

### CODE REVIEW

File: packages/design-system/src/components/input/test/validation.js [hunk 1]
```diff
@@ -52,21 +52,24 @@ describe('isInputValid - should validate input', () => {
...
+    ${'over maximum as float'}           | ${999.83}    | ${false}   | ${false} | ${10}        | ${undefined} | ${10}
+    ${'under minimum as string'}         | ${'-100'}    | ${false}   | ${false} | ${undefined} | ${0}         | ${0}
+    ${'over maximum as string'}          | ${'99999'}   | ${false}   | ${false} | ${10}        | ${undefined} | ${10}
+    ${'under minimum as string float'}   | ${'-100.12'} | ${false}   | ${false} | ${undefined} | ${0}         | ${0}
+    ${'over maximum as string float'}    | ${'999.83'}  | ${false}   | ${false} | ${10}        | ${undefined} | ${10}
+    ${'empty and allow empty'}           | ${''}        | ${true}    | ${false} | ${undefined} | ${undefined} | ${''}
+    ${'empty and disallow empty'}        | ${''}        | ${false}   | ${false} | ${undefined} | ${undefined} | ${null}
+    ${'minimum and maximum in range'}    | ${25}        | ${false}   | ${false} | ${100}       | ${0}         | ${25}
+    ${'minimum and maximum below range'} | ${-25}       | ${false}   | ${false} | ${100}       | ${0}         | ${0}
+    ${'minimum and maximum above range'} | ${125}       | ${false}   | ${false} | ${100}       | ${0}         | ${100}
```
Comment (mwritter) [inline packages/design-system/src/components/input/test/validation.js:72]: Lines 70, 71, and 73 were added - the other lines just needed formatting 👍

**Review by BrookeGraham — DISMISSED** [review]
Great find! 🎉

**Review by maxyinger — DISMISSED** [review]

**Review by sblinde — DISMISSED** [review]
🚀

**Review by sblinde — APPROVED** [review]
🐝  Looks good

**Pull request merged**